.github/workflows/debug_coverage.yml

```diff
@@ -30,25 +30,61 @@ jobs:
         shell: bash
         run: |
           sudo chown -R $(id -u):$(id -g) .
-      - uses: actions/checkout@v2.3.4
+      # Finds an associated PR (PR can be detected only on push and never on pull_request).
+      # WARNING !!! use in these ways only:
+      #   on push: steps.findPr.outputs.pr
+      #   on pull_request: github.event.pull_request.number
+      - name: Find the PR associated with this push on push trigger, if there is one.
+        uses: jwalton/gh-find-current-pr@v1
+        id: findPr
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      # Run always but once - w/o PR on push otherwise on pull_request.
+      # To avoid of issue https://github.com/coverallsapp/github-action/issues/55
+      # sources checkout should be based on 2nd action version and later.
+      - name: Sources checkout
+        if: ( steps.findPr.outputs.pr == false && github.event_name == 'push' ) ||
+            github.event_name == 'pull_request'
+        uses: actions/checkout@v2.3.4
         with:
           fetch-depth: 0
           submodules: recursive
-      - uses: ./.github/actions/environment
+      # Run always but once - w/o PR on push otherwise on pull_request.
```

Comment:
After some discussion with @avtikhon, it was decided to change the comment to something like::"Runs w/o PR on push otherwise on PR events."

Reply:
Sure, done.